pkg/plugin/v3/scaffolds/internal/templates/config/api/webhook_suite.go

```diff
@@ -0,0 +1,195 @@
+package api
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"sigs.k8s.io/kubebuilder/pkg/model/file"
+)
+
+var _ file.Template = &WebhookSuite{}
+var _ file.Inserter = &WebhookSuite{}
+
+// WebhookSuite scaffolds the webhook_suite.go file to setup the webhook test
+type WebhookSuite struct {
+	file.TemplateMixin
+	file.MultiGroupMixin
+	file.BoilerplateMixin
+	file.ResourceMixin
+
+	// BaseDirectoryRelativePath define the Path for the base directory when it is multigroup
+	BaseDirectoryRelativePath string
+}
+
+// SetTemplateDefaults implements file.Template
+func (f *WebhookSuite) SetTemplateDefaults() error {
+	if f.Path == "" {
+		if f.MultiGroup {
+			if f.Resource.Group != "" {
+				f.Path = filepath.Join("apis", "%[group]", "%[version]", "webhook_suite_test.go")
+			} else {
+				f.Path = filepath.Join("apis", "%[version]", "webhook_suite_test.go")
+			}
+		} else {
+			f.Path = filepath.Join("api", "%[version]", "webhook_suite_test.go")
+		}
+	}
+	f.Path = f.Resource.Replacer().Replace(f.Path)
+
+	f.TemplateBody = fmt.Sprintf(webhookTestSuiteTemplate,
+		file.NewMarkerFor(f.Path, addWebhookManagerMarker),
+	)
+
+	// If is multigroup the path needs to be ../../ since it has
+	// the group dir.
+	f.BaseDirectoryRelativePath = `"..", ".."`
+	if f.MultiGroup && f.Resource.Group != "" {
+		f.BaseDirectoryRelativePath = `"..", "..",".."`
+	}
+
+	return nil
+}
+
+const (
+	addWebhookManagerMarker = "webhook"
+)
+
+// GetMarkers implements file.Inserter
+func (f *WebhookSuite) GetMarkers() []file.Marker {
+	return []file.Marker{
+		file.NewMarkerFor(f.Path, addWebhookManagerMarker),
+	}
+}
+
+const (
+	addWebhookManagerCodeFragment = `err = (&%s{}).SetupWebhookWithManager(mgr)
+Expect(err).NotTo(HaveOccurred())
+
+`
+)
+
+// GetCodeFragments implements file.Inserter
+func (f *WebhookSuite) GetCodeFragments() file.CodeFragmentsMap {
+	fragments := make(file.CodeFragmentsMap, 2)
+
+	// Generate add webhookManager code fragments
+	addWebhookManager := make([]string, 0)
+	addWebhookManager = append(addWebhookManager, fmt.Sprintf(addWebhookManagerCodeFragment, f.Resource.Kind))
+
+	// Only store code fragments in the map if the slices are non-empty
+	if len(addWebhookManager) != 0 {
+		fragments[file.NewMarkerFor(f.Path, addWebhookManagerMarker)] = addWebhookManager
+	}
+
+	return fragments
+}
+
+const (
+	webhookTestSuiteTemplate = `
+package {{ .Resource.Version }}
+
+import (
+	"path/filepath"
+	"testing"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	admissionv1 "k8s.io/api/admission/v1"
```

Comment:
The api version used to scaffold the webhooks still v1betav1. so, would not better to use https://pkg.go.dev/k8s.io/api@v0.19.2/admission/v1beta1. 

also, if we change that then, not in the scaffold but in the code would be nice have a comment with a todo for we parametrize the version according to the version used to scaffold the webhooks. More info see: https://github.com/kubernetes-sigs/kubebuilder/pull/1644#discussion_r506385099

WDYT?

Reply:
Got it. So for now, I make it v1beta1 and add a TODO in the scaffold to change once we finalize the approach on how to handle it

Reply:
I think would be nice add value to the SetTemplate maybe. WDYT?

Reply:
Got it. Yes it would be nice when we have to handle both the version. Will do the changes.

Reply:
Done